testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,178 @@
+  // similar to one of the following:
+  //
+  //   - gcr.io/grpc-testing/java_worker:v3.2.19
+  //   - gcr.io/grpc-testing/java_worker@sha256:82b6360a84c19f23ed7ee9...
+  //
+  // The "latest" tag is automatically set by the registry, and there
+  // are no guarantees that it will point to a specific image. It should be
+  // avoided.
+  string container = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The kind which assigns the responsibilities of this container.
+  Kind kind = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  // The number of identical components that should be created with these
+  // settings. For example, a "client" component can be configured once and
+  // specify "2" replicas. Two identical versions will be available for tests.
+  //
+  // This field is ignored for driver components. It may eventually be supported
+  // on server components, but this is not yet available.
+  int32 replicas = 3;
+}
+
+message StartTestSessionRequest {
+  // The test scenarios to run as a part of the test session.
+  repeated grpc.testing.Scenario scenarios = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The component which orchestrates running the test scenarios amongst workers.
+  Component driver = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  // The components that are required as part of the test. Normally, this will
+  // involve at least one server and a number of clients.
+  repeated Component workers = 3;  // [(google.api.field_behavior) = REQUIRED];
+
+  // An optional location where the results should be written.
+  oneof result_destination {
+    // The name of the BigQuery table where results should be written.
+    string bq_results_table = 4;  // [(google.api.field_behavior) = OPTIONAL];
+
+    // The fully qualified GCS URI where a results file should be written.
+    string gcs_results_file_uri = 5;  // [(google.api.field_behavior) = OPTIONAL];
+  }
+ }
+
+message TestSessionResults {
+  // The results for each scenario.
+  repeated grpc.testing.ScenarioResult scenario_results = 1;
+
+  // The amount of time that this session lived, including all actions from
+  // scheduling to termination.
+  google.protobuf.Duration time_elapsed = 2;
+}
+
+message Event {
+  enum Kind {
+    // The event does not conform to the standard lifecycle.
+    UNKNOWN = 0;
+
+    // Currently waiting for resources to be provisioned.
+    QUEUE = 0;
+
+    // Creating, reserving or provisioning resources.
+    PROVISION = 1;
+
+    // Currently being executed on the provisioned resources.
+    RUN = 3;
+
+    // An error has occurred, preventing further events.
+    ERROR = 4;
+  }
+
+  // The kind of event.
+  Kind kind = 1;
+
+  // The time the event was recorded.
+  google.protobuf.Timestamp time = 2;
+
+  // The component which is the subject of this event, if applicable. When the
+  // event pertains to the entire TestSession, this field will be empty.
+  Component subject = 3;
```

Comment:
I made an **`Event`** applicable to both an entire test session and individual components.  This way, we can build up an event history log. This may help to quickly identify what is problematic.